core/src/main/scala/better/files/File.scala

```diff
@@ -152,8 +152,11 @@ class File private (val path: Path)(implicit val fileSystem: FileSystem = path.g
   )(implicit
       attributes: File.Attributes = File.Attributes.default,
       linkOptions: File.LinkOptions = File.LinkOptions.default
-  ): File =
-    (this / child).createIfNotExists(asDirectory, createParents)(attributes, linkOptions)
+  ): File = {
+    val destination = this / child
+    if (destination.isChildOf(this)) destination.createIfNotExists(asDirectory, createParents)(attributes, linkOptions)
```

Comment:
I am not quite sure if this is the right place since this method is used by other places too

Reply:
Ok.
So, should I add some tests to assure that this change will not affect any other codes using `createChild` method or do you have any idea where to place it?

Reply:
I think this check and throw should belong local to the unzip util

Reply:
I thought too for the first time, but now I think ``createChild`` should be responsible for validation while ``unzipTo`` should responsible for unzip because checking if file is inside directory is not unzip's concern. Besides, it feels strange if it is possible for ``createChild`` method to create file or directory in ancestor directory.

Reply:
Hmm I think you are right ..

Reply:
I agree with your concern that this change may affect codes using ``createChild`` method. So, if I write additional tests for methods using ``createChild``, chould you accept this PR?

Reply:
Yes